python/ray/rllib/optimizers/sample_batch.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from functools import reduce
+
+import numpy as np
+
+
+class SampleBatch(object):
+    """Wrapper around a dictionary with string keys and array-like values.
+
+    For example, {"obs": [1, 2, 3], "reward": [0, -1, 1]} is a batch of three
+    samples, each with an "obs" and "reward" attribute.
```

Comment:
Should Shuffle also be a method here?

Reply:
Hmm possibly. Right now multi gpu optimizer does a more efficient shuffling where it only permutes indices, which seems just as good.